staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresourcedefinition/strategy.go

```diff
@@ -259,3 +186,103 @@ func MatchCustomResourceDefinition(label labels.Selector, field fields.Selector)
 func CustomResourceDefinitionToSelectableFields(obj *apiextensions.CustomResourceDefinition) fields.Set {
 	return generic.ObjectMetaFieldsSet(&obj.ObjectMeta, true)
 }
+
+func dropDisabledFields(crdSpec, oldCrdSpec *apiextensions.CustomResourceDefinitionSpec) {
+	// if the feature gate is disabled, drop the feature.
+	if !utilfeature.DefaultFeatureGate.Enabled(apiextensionsfeatures.CustomResourceValidation) {
+		unversionedValidationInUse := oldCrdSpec != nil && oldCrdSpec.Validation != nil
+		versionedValidationInUse := versionSchemaInUse(oldCrdSpec)
+
+		if !unversionedValidationInUse && !versionedValidationInUse {
+			crdSpec.Validation = nil
+			for i := range crdSpec.Versions {
+				crdSpec.Versions[i].Schema = nil
+			}
+		}
+	}
+	if !utilfeature.DefaultFeatureGate.Enabled(apiextensionsfeatures.CustomResourceSubresources) {
+		unversionedSubresourceInUse := oldCrdSpec != nil && oldCrdSpec.Subresources != nil
+		versionedSubresourceInUse := versionSubresourceInUse(oldCrdSpec)
```

Comment:
same comment here, change to `subresourceInUse` and make it check both top-level and per-version subresources